droid-results/src/main/java/uk/gov/nationalarchives/droid/internal/api/ContainerApi.java

```diff
@@ -0,0 +1,126 @@
+import uk.gov.nationalarchives.droid.container.IdentifierEngine;
+import uk.gov.nationalarchives.droid.container.ole2.Ole2Identifier;
+import uk.gov.nationalarchives.droid.container.ole2.Ole2IdentifierEngine;
+import uk.gov.nationalarchives.droid.container.zip.ZipIdentifier;
+import uk.gov.nationalarchives.droid.container.zip.ZipIdentifierEngine;
+import uk.gov.nationalarchives.droid.core.interfaces.DroidCore;
+import uk.gov.nationalarchives.droid.core.interfaces.archive.ArchiveFormatResolver;
+import uk.gov.nationalarchives.droid.core.interfaces.archive.ArchiveFormatResolverImpl;
+import uk.gov.nationalarchives.droid.core.interfaces.archive.ContainerIdentifierFactory;
+import uk.gov.nationalarchives.droid.core.interfaces.archive.IdentificationRequestFactory;
+import uk.gov.nationalarchives.droid.core.interfaces.archive.ContainerIdentifierFactoryImpl;
+import uk.gov.nationalarchives.droid.core.interfaces.signature.SignatureFileException;
+
+public final class ContainerApi {
+
+    private final DroidCore droid;
+    private final Path containerSignature;
+
+    public ContainerApi(DroidCore droid, Path containerSignature) {
+        this.droid = droid;
+        this.containerSignature = containerSignature;
+    }
+
+    private ContainerSignatureFileReader signatureReader() {
+        ContainerSignatureFileReader reader = new ContainerSignatureFileReader();
+        reader.setFilePath(containerSignature.toAbsolutePath().toString());
+        return reader;
+    }
+
+    private IdentificationRequestFactory requestFactory() {
+        ContainerFileIdentificationRequestFactory factory = new ContainerFileIdentificationRequestFactory();
+        return factory;
+    }
+
+    private IdentifierEngine identifierEngine() {
+        ZipIdentifierEngine engine = new ZipIdentifierEngine();
+        engine.setRequestFactory(requestFactory());
+        return engine;
+    }
+
+    private ArchiveFormatResolver archiveFormatResolver() {
+        ArchiveFormatResolverImpl resolver = new ArchiveFormatResolverImpl();
+
+        return resolver;
+    }
+
+    private ContainerIdentifierFactory identifierFactory() {
+        ContainerIdentifierFactoryImpl identifierFactory = new ContainerIdentifierFactoryImpl();
+        return identifierFactory;
+    }
+
+    public ZipIdentifier zipIdentifier() {
+        ZipIdentifier zip = new ZipIdentifier();
+        zip.setContainerType("ZIP");
+        zip.setContainerIdentifierFactory(identifierFactory());
+        zip.setContainerFormatResolver(archiveFormatResolver());
+        zip.setDroidCore(droid);
+        zip.setIdentifierEngine(identifierEngine());
+        zip.setSignatureReader(signatureReader());
+
+        try {
+            zip.init();
+        } catch (SignatureFileException e) {
+            throw new RuntimeException("Unable to init zip identifier", e);
+        }
+        return zip;
+    }
+
+    private Ole2IdentifierEngine ole2IdentifierEngine() {
```

Comment:
Is this one line method necessary?

Reply:
Maybe not. I just what to have it same. I will remove.